### PR TITLE
Database Migration to create tables for resources

### PR DESCRIPTION
**Pull request by SM43**
This migration create resources, catalogs, resource_versions
tables and add the foreign key constraint.

Signed-off-by: Shivam Mukhade <smukhade@redhat.com>

### CODE REVIEW

File: api/cmd/db/migration.go [hunk 1]
```diff
@@ -14,32 +14,63 @@ func Migrate(api *app.APIConfig) error {
 
 	// NOTE: when writing a migration for a new table, add the same in InitSchema
 	migration := gormigrate.New(api.DB(), gormigrate.DefaultOptions, []*gormigrate.Migration{
-		// NOTE: Add Migrations Here
+		{
+			ID: "202006071000",
+			Migrate: func(tx *gorm.DB) error {
+				if err := tx.AutoMigrate(&model.Tag{}, &model.Catalog{}, &model.Resource{}, &model.ResourceVersion{}).Error; err != nil {
```
Comment (PuneetPunamiya) [inline api/cmd/db/migration.go:20]: Are we missing &model.category ??

Comment (SM43) [reply api/cmd/db/migration.go:20]: Migration will run for creating new tables or columns in existing tables

**Pull request merged**